pynamodb/connection/base.py

```diff
@@ -463,10 +465,15 @@ def _handle_binary_attributes(data):
                 for attr in six.itervalues(item):
                     _convert_binary(attr)
         if RESPONSES in data:
-            for item_list in six.itervalues(data[RESPONSES]):
```

Comment:
Could you elaborate on the purpose of this change? Probably worth adding comments to the code as well.

Also, is there a chance that in the future the type here won't be concretely `list`? Checking for a concrete list type like this looks superficially dangerous without understanding full reasoning.

Reply:
batch operations return `Responses` as a dict (which is why it iterates over the `itervalues`) whereas transaction operations return them as a list. Is there some less specific type check you'd prefer I do?

Reply:
No specific preference without investigating boto, but the potential concern is just that relying on a concrete type could potentially cause breakage in the future if all boto is guaranteeing is iterable for example. My comment was more of a general concern around relying on a concrete type, than a claim that it's actually wrong. If it seems safe that this won't break with future boto upgrades, we're good.

*edit*:  Looking at the implementation, I guess this is just the raw JSON response without any boto steps in between, so this shouldn't break unless DynamoDB changes its behavior so it's fine. I commented before realizing this was a parse of a raw response.